### PR TITLE
docs(design/autoware-interfaces): add map interface

### PR DESCRIPTION
**Pull request by cyn-liu**
## Description

Add map interface document in [design/Autoware interfaces](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/).

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by mitsudome-r** [comment]
@n-patiphon Could you assign someone from mapping WG to review this PR?

**Comment by n-patiphon** [comment]
@mitsudome-r I will bring this up in the meeting tonight.

File: docs/design/autoware-interfaces/components/map.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+How to create maps reference [Creating maps for Autoware](../../../how-to-guides/creating-maps-for-autoware.md).
+
+## Outputs
+
+### Point cloud map
+
+Loads pointcloud file and publishes the maps to the other Autoware nodes in various configurations. Currently, it supports the following types:
+
+- Raw pointcloud map (sensor_msgs/msg/PointCloud2)
+- Downsampled pointcloud map (sensor_msgs/msg/PointCloud2)
```
Comment (mitsudome-r) [inline docs/design/autoware-interfaces/components/map.md:18]: I think downsampled maps are mainly for visualization so you might want to mention that in the documentation

File: docs/design/autoware-interfaces/components/map.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Map
+
+![Node diagram](./images/Map-Bus-ODD-Architecture.drawio.svg)
+
+## Inputs
+
+Autoware relies on high-definition point cloud maps and vector maps of the driving environment to perform various tasks. Before launch Autoware, you need to load the pre-created map files.
```
Comment (n-patiphon) [inline docs/design/autoware-interfaces/components/map.md:7]:
``` suggestion
Autoware relies on high-definition point cloud maps and vector maps of the driving environment to perform various tasks. Before launching Autoware, you need to load the pre-created map files.
```

File: docs/design/autoware-interfaces/components/map.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+How to create maps reference [Creating maps for Autoware](../../../how-to-guides/creating-maps-for-autoware.md).
```
Comment (n-patiphon) [inline docs/design/autoware-interfaces/components/map.md:9]:
``` suggestion
Refer to [Creating maps for Autoware](../../../how-to-guides/creating-maps-for-autoware.md) on how to create maps.
```

File: docs/design/autoware-interfaces/components/map.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## Outputs
+
+### Point cloud map
+
+Loads pointcloud file and publishes the maps to the other Autoware nodes in various configurations. Currently, it supports the following types:
```
Comment (n-patiphon) [inline docs/design/autoware-interfaces/components/map.md:15]:
``` suggestion
It loads point cloud files and publishes the maps to the other Autoware nodes in various configurations. Currently, it supports the following types:
```

File: docs/design/autoware-interfaces/components/map.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+- Raw pointcloud map (sensor_msgs/msg/PointCloud2)
```
Comment (n-patiphon) [inline docs/design/autoware-interfaces/components/map.md:17]:
``` suggestion
- Raw point cloud map (sensor_msgs/msg/PointCloud2)
```

File: docs/design/autoware-interfaces/components/map.md (hunk 1 above)
Comment (n-patiphon) [inline docs/design/autoware-interfaces/components/map.md:18]:
```suggestion
- Downsampled point cloud map (sensor_msgs/msg/PointCloud2)
```

File: docs/design/autoware-interfaces/components/map.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+- Downsampled pointcloud map (sensor_msgs/msg/PointCloud2)
+- Partial pointcloud map loading via ROS 2 service (autoware_map_msgs/srv/GetPartialPointCloudMap)
```
Comment (n-patiphon) [inline docs/design/autoware-interfaces/components/map.md:19]:
```suggestion
- Partial point cloud map loading via ROS 2 service (autoware_map_msgs/srv/GetPartialPointCloudMap)
```

File: docs/design/autoware-interfaces/components/map.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+- Differential pointcloud map loading via ROS 2 service (autoware_map_msgs/srv/GetDifferentialPointCloudMap)
```
Comment (n-patiphon) [inline docs/design/autoware-interfaces/components/map.md:20]:
```suggestion
- Differential point cloud map loading via ROS 2 service (autoware_map_msgs/srv/GetDifferentialPointCloudMap)
```

File: docs/design/autoware-interfaces/components/map.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+### Lanlet2 map
+
+Loads Lanelet2 file and publishes the map data as `autoware_auto_mapping_msgs/msg/HADMapBin` message. The lan/lon coordinates is projected into the MGRS coordinates.
```
Comment (n-patiphon) [inline docs/design/autoware-interfaces/components/map.md:24]:
``` suggestion
It loads a Lanelet2 file and publishes the map data as `autoware_auto_mapping_msgs/msg/HADMapBin` message. The lan/lon coordinates are projected onto the MGRS coordinates.
```

File: docs/design/autoware-interfaces/components/map.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+- autoware_auto_mapping_msgs/msg/HADMapBin
+  - std_msgs/Header header
+  - string version_map_format
+  - string version_map
+  - string name_map
+  - uint8[] data
+
+### Lanlet2 map visualization
+
+visualize `autoware_auto_mapping_msgs/HADMapBin` messages in `Rviz`.
```
Comment (n-patiphon) [inline docs/design/autoware-interfaces/components/map.md:35]:
``` suggestion
Visualize `autoware_auto_mapping_msgs/HADMapBin` messages in `Rviz`.
```

**Review by n-patiphon — COMMENTED** [review]
I'm sorry for the late review.

I made a few grammatical suggestions. They are mostly minor, but it would be nice if you could fix them.

Regarding the structure, I think the text under `Inputs` feels like an `Overview` to me. I guess you could probably modify it to something like:

```
# Map

## Overview

Autoware relies on high-definition point cloud maps and vector maps of the driving environment to perform various tasks. Before launching Autoware, you need to load the pre-created map files.
...
The Map server handles the maps loading and publishing.

## Inputs

- Point cloud maps (`.pcd`)
- Lanelet2 maps (`.osm`)

Refer to [Creating maps for Autoware](../../../how-to-guides/creating-maps-for-autoware.md) on how to create maps.

```

**Comment by liuXinGangChina** [comment]
Hi,mits-san. @mitsudome-r 
i look into the spell-check error which block the merge,it is about 'lanlet' 
i think it is a common knowledge among autoware-developer
so can you help us to merge this pr

**Comment by n-patiphon** [comment]
@liuXinGangChina Hi, there. My apologies for intruding here (I got a notification).
I think `lanlet` is a typo. The correct word would be `lanelet` or `Lanelet`.
If you check the [spelling list](https://raw.githubusercontent.com/tier4/autoware-spell-check-dict/main/.cspell.json), lanelet is included. So, if you fix that the CI would likely pass.

File: docs/design/autoware-interfaces/components/map.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+It loads a Lanelet2 file and publishes the map data as `autoware_auto_mapping_msgs/msg/HADMapBin` message. The lan/lon coordinates are projected onto the MGRS coordinates.
+
+- autoware_auto_mapping_msgs/msg/HADMapBin
+  - std_msgs/Header header
+  - string version_map_format
+  - string version_map
+  - string name_map
+  - uint8[] data
+
+### Lanlet2 map visualization
```
Comment (mitsudome-r) [inline docs/design/autoware-interfaces/components/map.md:38]:
```suggestion
### Lanelet2 map visualization
```

File: docs/design/autoware-interfaces/components/map.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+### Point cloud map
+
+It loads point cloud files and publishes the maps to the other Autoware nodes in various configurations. Currently, it supports the following types:
+
+- Raw point cloud map (sensor_msgs/msg/PointCloud2)
+- Downsampled point cloud map (sensor_msgs/msg/PointCloud2)
+- Partial point cloud map loading via ROS2 service (autoware_map_msgs/srv/GetPartialPointCloudMap)
+- Differential point cloud map loading via ROS2 service (autoware_map_msgs/srv/GetDifferentialPointCloudMap)
+
+### Lanlet2 map
```
Comment (mitsudome-r) [inline docs/design/autoware-interfaces/components/map.md:27]:
```suggestion
### Lanelet2 map
```

File: docs/design/autoware-interfaces/components/map.md (hunk 1 above)
Comment (cyn-liu) [reply docs/design/autoware-interfaces/components/map.md]: The pointcloud_map_loader node  publish the original point cloud and downsampled point cloud

**Comment by cyn-liu** [comment]
> @liuXinGangChina Hi, there. My apologies for intruding here (I got a notification). I think `lanlet` is a typo. The correct word would be `lanelet` or `Lanelet`. If you check the [spelling list](https://raw.githubusercontent.com/tier4/autoware-spell-check-dict/main/.cspell.json), lanelet is included. So, if you fix that the CI would likely pass.

Thanks for your reply, I have modified the word `lanlet` and there will be no spelling errors now, but a new error `Markdown Link Check` was discovered,  this errors does not belong to the file I submitted.

**Comment by n-patiphon** [comment]
Hi @cyn-liu!
I checked the link, and it seems to be MAP IV Inc. URL. I believe it's already fixed in this [PR](https://github.com/autowarefoundation/autoware-documentation/pull/349). So, if you rebase or merge the lastest `main`, the problem should disappear.

**Comment by cyn-liu** [comment]
> Hi @cyn-liu! I checked the link, and it seems to be MAP IV Inc. URL. I believe it's already fixed in this [PR](https://github.com/autowarefoundation/autoware-documentation/pull/349). So, if you rebase or merge the latest `main`, the problem should disappear.

I already rebase this PR.

**Comment by liuXinGangChina** [comment]
Hi mits-san @mitsudome-r , since @cyn-liu has already rebased can you help us to merge it.

**Review by mitsudome-r — APPROVED** [review]
LGTM

**Pull request merged**